### PR TITLE
Merge homebrew 1.3.0

### PR DESCRIPTION
**Pull request by jonchang**
Merge branch homebrew/master into linuxbrew/master

Conflicts:
      Library/Homebrew/formula_support.rb
      Library/Homebrew/test/cleanup_spec.rb

### CODE REVIEW

File: Library/Homebrew/test/cleanup_spec.rb [hunk 1]
```diff
@@ -58,13 +52,13 @@
...
       it "doesn't remove any kegs" do
-        shutup { described_class.cleanup_formula f2 }
+        described_class.cleanup_formula f2
         expect(f1.installed_kegs.size).to eq(2)
       end
 
       it "lists the unremovable kegs" do
         shutup { described_class.cleanup_formula f2 }
-        expect(described_class.unremovable_kegs).to contain_exactly(f1.installed_kegs.sort_by(&:to_s)[0])
+        expect(described_class.unremovable_kegs).to contain_exactly(f1.installed_kegs[0])
```
Comment (sjackman) [inline Library/Homebrew/test/cleanup_spec.rb]: This test may fail due to the file system sorting either chronologically or lexicographically. Try
```ruby
expect(described_class.unremovable_kegs).to contain_exactly(f1.installed_kegs.sort_by(&:to_s)[0])
```

**Review by sjackman — CHANGES_REQUESTED** [review]

**Comment by jonchang** [comment]
Alright, fixes made. This should be ready to go once 1.2.6 is tagged and we see no issues.

**Comment by sjackman** [comment]
```
Error: Calling fails_with :llvm is deprecated!
There is no replacement.
/home/ubuntu/brew/Library/Taps/homebrew/homebrew-core/Formula/gcc.rb:63:in `<class:Gcc>'
Please report this to the homebrew/core tap!
/usr/bin/gcc: 4.8.4
glibc: N/A
==> FAILED
```
Interesting that this affects CircleCI but not TravisCI, can't say why.
@jonchang Want to submit a PR to Linuxbrew/core to fix it?

**Comment by sjackman** [comment]
I've opened PR https://github.com/Linuxbrew/homebrew-core/pull/3610 to fix the above error. After that PR is merged, the CI tests for this PR can be retriggered. Assuming they pass, this PR can be merged.

**Review by sjackman — APPROVED** [review]

**Comment by sjackman** [comment]
Merged. Thanks, Jon!

**Pull request closed without merging**